src/api/setAwayStatus.js

```diff
@@ -0,0 +1,11 @@
+/* @flow strict-local */
+import type { ApiResponseSuccess, Auth } from './apiTypes';
+import { apiPost, objectToParams } from './apiFetch';
+
+type SetAwayStatusParams = {
+  away?: boolean,
+  status_text?: string,
+};
+
+export default (auth: Auth, params: SetAwayStatusParams): Promise<ApiResponseSuccess> =>
+  apiPost(auth, 'users/me/status', objectToParams(params));
```

Comment:
What's the reasoning for using `objectToParams` here?

I believe it should have no effect at all -- neither of these properties are permitted to be an array, or `undefined`. So it seems simpler to just say `params`.

Reply:
The function is used for its ability to remove keys with `undefined` values.

Reply:
This isn't much of a reply. I wrote above:

> I believe it should have no effect at all -- neither of these properties are permitted to be an array, or `undefined`.

If that's mistaken, would you explain why? If not, why do you take steps to remove keys with `undefined` values?

Reply:
Aha, I see in a later commit that you do pass it params objects that have `undefined` values. I was confused to see that Flow accepted that; it turns out that the documented semantics of `?:` properties is [like so](https://flow.org/en/docs/types/objects/#toc-optional-object-type-properties):
> In addition to their set value type, these optional properties can either be `void` or omitted altogether. However, they cannot be `null`.

So indeed this is doing something useful, then.

(This reply remains not much of a reply.)